pkg/hwinfo/hwinfo_other.go

```diff
@@ -194,3 +212,234 @@ func listDisplays() ([]*monitorInfo, error) {
+
+			t, err := strconv.ParseInt(val, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+
+			c.Siblings = int32(t)
+		}
+	}
+
+	if c.CPU >= 0 {
+		err := finishCPUInfo(&c)
+		if err != nil {
+			return nil, err
+		}
+		cpus = append(cpus, c)
+	}
+
+	sorted := make(map[string]*cpuInfo)
+
+	for _, cpu := range cpus {
+		if _, found := sorted[cpu.PhysicalID]; !found {
+			info := &cpuInfo{
+				manufacturer:      cpu.VendorID,
+				manufacturingInfo: fmt.Sprintf("Model %s Family %s Stepping %d", cpu.Model, cpu.Family, cpu.Stepping),
+				description:       cpu.ModelName,
+				coreCount:         fmt.Sprintf("%d", cpu.Cores),
+				coreEnabled:       fmt.Sprintf("%d", cpu.Cores),
+				threadCount:       fmt.Sprintf("%d", cpu.Siblings),
+			}
+
+			sorted[cpu.PhysicalID] = info
+		}
+	}
+
+	var cInfo []cpuInfo
+	for _, cpu := range sorted {
+		cInfo = append(cInfo, *cpu)
+	}
+
+	return cInfo, nil
+}
+
+func finishCPUInfo(c *cpuStat) error {
+	var lines []string
+	var err error
+	var value float64
+
+	if len(c.CoreID) == 0 {
+		lines, err = readLines(sysCPUPath(c.CPU, "topology/core_id"))
+		if err == nil {
+			c.CoreID = lines[0]
+		}
+	}
+
+	// override the value of c.Mhz with cpufreq/cpuinfo_max_freq regardless
+	// of the value from /proc/cpuinfo because we want to report the maximum
+	// clock-speed of the CPU for c.Mhz, matching the behaviour of Windows
+	lines, err = readLines(sysCPUPath(c.CPU, "cpufreq/cpuinfo_max_freq"))
+	// if we encounter errors below such as there are no cpuinfo_max_freq file,
+	// we just ignore. so let Mhz is 0.
+	if err != nil {
+		return nil
+	}
+	value, err = strconv.ParseFloat(lines[0], 64)
+	if err != nil {
+		return nil
+	}
+	c.Mhz = value / 1000.0 // value is in kHz
+	if c.Mhz > 9999 {
+		c.Mhz = c.Mhz / 1000.0 // value in Hz
+	}
+	return nil
+}
+
+// ReadLines reads contents from a file and splits them by new lines.
+// A convenience wrapper to ReadLinesOffsetN(filename, 0, -1).
+func readLines(filename string) ([]string, error) {
+	return readLinesOffsetN(filename, 0, -1)
+}
+
+// ReadLines reads contents from file and splits them by new line.
+// The offset tells at which line number to start.
+// The count determines the number of lines to read (starting from offset):
+//   n >= 0: at most n lines
+//   n < 0: whole file
+func readLinesOffsetN(filename string, offset uint, n int) ([]string, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return []string{""}, err
+	}
+	defer f.Close()
+
+	var ret []string
+
+	r := bufio.NewReader(f)
+	for i := 0; i < n+int(offset) || n < 0; i++ {
+		line, err := r.ReadString('\n')
+		if err != nil {
+			break
+		}
+		if i < int(offset) {
+			continue
+		}
+		ret = append(ret, strings.Trim(line, "\n"))
+	}
+
+	return ret, nil
+}
+
+func sysCPUPath(cpu int32, relPath string) string {
+	return getEnv("HOST_SYS", "/sys", fmt.Sprintf("devices/system/cpu/cpu%d", cpu), relPath)
+}
+
+func getEnv(key string, dfault string, combineWith ...string) string {
```

Comment:
The name seems inaccurate. The function does more than just getting a value from environment

Reply:
This is actually the function from `gopsutil` package.
We already have copied implementations into other packages. Probably it worth moving such function to `common` package

Reply:
done